language-design.md

```diff
@@ -28,6 +83,14 @@ you know a poorly designed language? What are the symptoms?
 
 **Response**
 
+One noticeable characteristic of a well-designed langauge is that it makes space
+for growth. User are able to grow the language themselves, following the bazaar
+metaphor in the reading from Steele. Another similar idea is that users are
+able to build things in the language that looks like things that are built into
+the language to give them a sense ownership. If the language is inflexible, it
+makes it hard for the user to make things. Furthermore, if it's built large and
+without collaboration, it is likely designed in a way that is not what the users
+would like.
```

Comment:
A lot of why Java is successful, in my opinion, has to do with four factors:

1. It's taught very heavily in most schools (not Mudd, but many others). It's their intro, intermediate, and advanced language.
2. It runs pretty much anywhere. One of the greatest banes of software development is portability, and Java does a really good job of abstracting away OS idiosyncracies.
3. This is entirely speculative, and I might be getting my history wrong, but I think it hopped on the OOP train pretty early when people weren't really used to it yet. We can see now how popular and (relatively) easy to understand OOP is, and I would assume that it was appealing to a lot of people.
4. As for why it remains successful, at this point it's in a lot of systems and server code, to the point where rewriting it in another language is unpleasant or unfeasible, so people are going to stick with it for that alone. I don't think that technical debt is necessarily a good reason, but its a reason.

That being said, I've also found that there are a lot of people (myself included, so bias warning) who don't like Java. I think a lot of this is people who used old versions of Java found themselves really bashing their heads against the wall to work with it; I learned on Java 5 and found it atrocious, but Java 8 has been quite pleasant when I've used it.